### PR TITLE
gimx: unstable -> 8.0

### PR DESCRIPTION
**Pull request by bb2020**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/gimx/default.nix [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+  gimx-pdp = false;
   gimx-config = fetchFromGitHub {
     owner = "matlo";
     repo = "GIMX-configurations";
     rev = "c20300f24d32651d369e2b27614b62f4b856e4a0";
-    sha256 = "02wcjk8da188x7y0jf3p0arjdh9zbb0lla3fxdb28b1xyybfvx5p";
+    hash = "sha256-t/Ttlvc9LCRW624oSsFaP8EmswJ3OAn86QgF1dCUjAs=";
+  };
+  gimx-patch = fetchpatch {
+    url = "https://github.com/matlo/GIMX/pull/705.patch";
```
Comment (SuperSandro2000) [inline pkgs/games/gimx/default.nix:12]: That is not merged, please pull the individual commits

File: pkgs/games/gimx/default.nix [hunk 2]
```diff
@@ -18,57 +21,31 @@ in stdenv.mkDerivation rec {
     owner = "matlo";
     repo = "GIMX";
     rev = "58d2098dce75ed4c90ae649460d3a7a150f4ef0a";
+    hash = "sha256-/9EYBrqHooXsB4gVfS/bBKyD8360QXOuYCGMjLHYbRY=";
     fetchSubmodules = true;
-    sha256 = "05kdv2qqr311c2p76hdlgvrq7b04vcpps5c80zn8b8l7p831ilgz";
```
Comment (SuperSandro2000) [inline pkgs/games/gimx/default.nix:22]: Before it was more correct as a change to fetchSubmodules invalidates the hash

File: pkgs/games/gimx/default.nix [hunk 3]
```diff
@@ -1,13 +1,16 @@
-{ stdenv, lib, fetchFromGitHub, fetchpatch
-, makeWrapper, curl, libusb1, xorg, libxml2
-, ncurses5, bluez, libmhash, gimxPdpGamepad ? false }:
+{ stdenv, lib, fetchFromGitHub, fetchpatch, makeWrapper, curl, libusb1, bluez, libxml2, ncurses5, libmhash, xorg }:
 
 let
+  gimx-pdp = false;
```
Comment (SuperSandro2000) [inline pkgs/games/gimx/default.nix:4]: If this is not configurable, we might as well drop the code related to it.

File: pkgs/games/gimx/default.nix [hunk 4]
```diff
@@ -18,57 +21,31 @@ in stdenv.mkDerivation rec {
...
-  ];
-
+  env.NIX_CFLAGS_COMPILE = "-Wno-error";
   nativeBuildInputs = [ makeWrapper ];
-  buildInputs = [
-    curl libusb1 bluez libxml2 ncurses5 libmhash
-    xorg.libX11 xorg.libXi xorg.libXext
-  ];
+  buildInputs = [ curl libusb1 bluez libxml2 ncurses5 libmhash xorg.libX11 xorg.libXi ];
+  patches = [ ./conf.patch gimx-patch ];
```
Comment (SuperSandro2000) [inline pkgs/games/gimx/default.nix:31]: The patch here should be inlined.

File: pkgs/games/gimx/default.nix (hunk 3 above)
Comment (bb2020) [reply pkgs/games/gimx/default.nix:4]: We may use it in the future.

Comment (SuperSandro2000) [reply pkgs/games/gimx/default.nix:4]: Then this is currently dead code which we don't want to accumulate. We always have the git history to recover it.

File: pkgs/games/gimx/default.nix [hunk 5]
```diff
@@ -1,103 +1,50 @@
...
 
-  env.NIX_CFLAGS_COMPILE = toString [
-    # Needed with GCC 12
-    "-Wno-error=address"
-    "-Wno-error=deprecated-declarations"
-    "-Wno-error=use-after-free"
-  ];
-
   installPhase = ''
-    runHook preInstall
```
Comment (SuperSandro2000) [inline pkgs/games/gimx/default.nix:67]: The runHooks should be kept

File: pkgs/games/gimx/default.nix [hunk 6]
```diff
@@ -1,103 +1,54 @@
...
+  version = "8.0";
 
   src = fetchFromGitHub {
+    fetchSubmodules = true;
     owner = "matlo";
     repo = "GIMX";
-    rev = "58d2098dce75ed4c90ae649460d3a7a150f4ef0a";
-    fetchSubmodules = true;
-    sha256 = "05kdv2qqr311c2p76hdlgvrq7b04vcpps5c80zn8b8l7p831ilgz";
+    rev = "v${version}";
```
Comment (SuperSandro2000) [inline pkgs/games/gimx/default.nix:19]:
```suggestion
    rev = "v${version}";
    fetchSubmodules = true;
```

File: pkgs/games/gimx/default.nix [hunk 7]
```diff
@@ -1,103 +1,54 @@
...
+    hash = "sha256-t/Ttlvc9LCRW624oSsFaP8EmswJ3OAn86QgF1dCUjAs=";
   };
 
 in stdenv.mkDerivation rec {
   pname = "gimx";
-  version = "unstable-2021-08-31";
+  version = "8.0";
 
   src = fetchFromGitHub {
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/games/gimx/default.nix:16]:
```suggestion
```

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**